### PR TITLE
Reduce usage of InstantiatorBackedObjectFactory

### PR DESCRIPTION
**Pull request by asodja**
This removes some usages of `InstantiatorBackedObjectFactory`, there are 2 more left but they are more complicated.

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: subprojects/core/src/main/java/org/gradle/groovy/scripts/DefaultScript.java [hunk 1]
```diff
@@ -87,12 +87,13 @@ public void init(final Object target, ServiceRegistry services) {
...
                 FileResolver resolver = fileLookup.getFileResolver(sourceFile.getParentFile());
                 FileCollectionFactory fileCollectionFactoryWithBase = fileCollectionFactory.withResolver(resolver);
+                ObjectFactory objectFactory = services.get(ObjectFactory.class);
                 fileOperations = DefaultFileOperations.createSimple(resolver, fileCollectionFactoryWithBase, services);
                 deprecatedProcessOperations = new DeprecatedProcessOperations(services.get(ExecFactory.class).forContext()
                     .withFileResolver(resolver)
                     .withFileCollectionFactory(fileCollectionFactoryWithBase)
                     .withInstantiator(instantiator)
-                    .withObjectFactory(new InstantiatorBackedObjectFactory(instantiator))
+                    .withObjectFactory(objectFactory)
```
Comment (asodja) [inline subprojects/core/src/main/java/org/gradle/groovy/scripts/DefaultScript.java:96]: 💭 We got rid of this `new InstantiatorBackedObjectFactory(instantiator)`

File: subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaDebugOptions.java [hunk 2]
```diff
@@ -35,17 +34,12 @@ public class DefaultJavaDebugOptions implements JavaDebugOptions {
...
-        this.host = objectFactory.property(String.class);
-        this.port = objectFactory.property(Integer.class).convention(5005);
-        this.server = objectFactory.property(Boolean.class).convention(true);
-        this.suspend = objectFactory.property(Boolean.class).convention(true);
-    }
-
-    public DefaultJavaDebugOptions() {
-        // Ugly, but there are a few places where we need to instantiate a JavaDebugOptions and a regular ObjectFactory service
-        // is not available.
-        this(new InstantiatorBackedObjectFactory(DirectInstantiator.INSTANCE));
```
Comment (asodja) [inline subprojects/core/src/main/java/org/gradle/process/internal/DefaultJavaDebugOptions.java:48]: 💭 We got rid of this usage of `InstantiatorBackedObjectFactory`

**Comment by asodja** [comment]
@bot-gradle test APT

File: subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecActionFactory.java [hunk 3]
```diff
@@ -155,8 +155,7 @@ public EffectiveJavaForkOptions toEffectiveJavaForkOptions(JavaForkOptionsIntern
         // NOTE: We do not want/need a decorated version of JvmOptions or JavaDebugOptions because
         // these immutable instances are held across builds and will retain classloaders/services in the decorated object
         DefaultFileCollectionFactory fileCollectionFactory = new DefaultFileCollectionFactory(fileResolver, DefaultTaskDependencyFactory.withNoAssociatedProject(), new DefaultDirectoryFileTreeFactory(), nonCachingPatternSetFactory, PropertyHost.NO_OP, FileSystems.getDefault());
-        ObjectFactory objectFactory = new InstantiatorBackedObjectFactory(DirectInstantiator.INSTANCE);
```
Comment (asodja) [inline subprojects/core/src/main/java/org/gradle/process/internal/DefaultExecActionFactory.java:158]: 💭 We got rid of this instance of `InstantiatorBackedObjectFactory`

**Comment by asodja** [comment]
@bot-gradle clean

**Review by big-guy — APPROVED** [review]

**Comment by asodja** [comment]
@bot-gradle test this

**Review by lptr — APPROVED** [review]
LGTM.

**Pull request merged**